apps/token/src/lib/url-connector.ts

```diff
@@ -0,0 +1,151 @@
+        params[0].gasLimit = params[0].gas;
+        delete params[0].gas;
+        // If from is present on eth_sendTransaction it errors, removing it makes the library set
+        // from as the connected wallet which works fine
+        delete params[0].from;
+        const req = ethers.providers.JsonRpcProvider.hexlifyTransaction(
+          params[0]
+        );
+        // Hexlify sets the gasLimit property to be gas again and send transaction requires gasLimit
+        req['gasLimit'] = req['gas'];
+        delete req['gas'];
+
+        if (!this.signer) {
+          throw new Error('No signer');
+        }
+
+        // Send the transaction
+        const tx = await this.signer.sendTransaction(req);
+        result = tx.hash;
+      } else {
+        // All other transactions the base class works for
+        result = await super.send(method, params);
+      }
+      console.debug('result received', method, params, result);
+      if (isCallbackForm) {
+        callback(null, { result });
+      } else {
+        return result;
+      }
+    } catch (error) {
+      console.error(error);
+      if (isCallbackForm) {
+        callback(error, null);
+      } else {
+        throw error;
+      }
+    }
+  }
+}
+
+type url = string | ConnectionInfo;
+
+export class Url extends Connector {
+  /** {@inheritdoc Connector.provider} */
+  public override provider: Eip1193Bridge | undefined;
+
+  private eagerConnection?: Promise<void>;
+  private url: url;
+
+  /**
+   * @param url - An RPC url.
+   * @param connectEagerly - A flag indicating whether connection should be initiated when the class is constructed.
+   */
+  constructor(actions: Actions, url: url, connectEagerly = false) {
+    super(actions);
+
+    if (connectEagerly && typeof window === 'undefined') {
+      throw new Error(
+        'connectEagerly = true is invalid for SSR, instead use the activate method in a useEffect'
+      );
+    }
+
+    this.url = url;
+
+    if (connectEagerly) void this.activate();
+  }
+
+  private async isomorphicInitialize() {
+    if (this.eagerConnection) return this.eagerConnection;
+
+    await (this.eagerConnection = import('@ethersproject/providers')
+      .then(({ JsonRpcProvider }) => JsonRpcProvider)
+      .then((JsonRpcProvider) => {
+        const provider = new JsonRpcProvider(this.url);
+        const privateKey = Wallet.fromMnemonic(
+          ENV.ethWalletMnemonic,
+          `m/44'/60'/0'/0/0`
+        ).privateKey;
+        const signer = new Wallet(privateKey, provider);
+        this.provider = new CustomizedBridge(signer, provider);
+        this.actions.update({ accounts: [signer.address], chainId: 1440 });
+      }));
+  }
+
+  /** {@inheritdoc Connector.activate} */
+  public async activate(): Promise<void> {
+    this.actions.startActivation();
+
+    await this.isomorphicInitialize();
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return this.provider!.request({ method: 'eth_chainId' })
```

Comment:
question (non-blocking): any reason for the mix of async and Promises in this file? not sure, i usually find it more readable when it's not mixed up